src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,204 @@
+    svg.setAttribute("viewBox", `0 0 ${XMLwidth} ${XMLheight}`);
+    $container.append(svg);
+    svg.setAttribute('class', 'textSelectionSVG');
+    svg.setAttribute('preserveAspectRatio', 'none');
+    $(svg).css({
+      "width": "100%",
+      "position": "absolute",
+      "height": "100%",
+      "top": "0",
+      "left": "0",
+    });
+
+    $(XMLpage).find("PARAGRAPH").each((i, paragraph) => {
+      // adding text element for each paragraph in the page
+      const paragSvg = document.createElementNS("http://www.w3.org/2000/svg", this.svgParagraphElement);
+      paragSvg.setAttribute("class", "BRparagElement");
+      const words = $(paragraph).find("WORD");
+      let wordHeightMax = 0;
+
+      for(let i = 0; i < words.length; i++) {
+        // adding tspan for each word in paragraph
+        const currWord = words[i];
+        // eslint-disable-next-line no-unused-vars
+        const [left, bottom, right, top] = $(currWord).attr("coords").split(',').map(parseFloat);
+        const wordHeight = bottom - top;
+        if(wordHeight > wordHeightMax) wordHeightMax = wordHeight;
+
+        const wordTspan = document.createElementNS("http://www.w3.org/2000/svg", this.svgWordElement);
+        wordTspan.setAttribute("class", "BRwordElement");
+        wordTspan.setAttribute("x", left.toString());
+        wordTspan.setAttribute("y", bottom.toString());
+        wordTspan.setAttribute("textLength", (right - left).toString());
+        wordTspan.setAttribute("lengthAdjust", "spacingAndGlyphs");
+        wordTspan.textContent = currWord.textContent;
+        paragSvg.appendChild(wordTspan);
+
+        // adding spaces after words except at the end of the paragraph
+        if(i < words.length - 1){
+          const nextWord = words[i + 1];
+          // eslint-disable-next-line no-unused-vars
+          const [leftNext, bottomNext, rightNext, topNext] = $(nextWord).attr("coords").split(',').map(parseFloat);
+          const spaceTspan = document.createElementNS("http://www.w3.org/2000/svg", this.svgWordElement);
+          spaceTspan.setAttribute("class", "BRwordElement");
+          spaceTspan.setAttribute("x", right.toString());
+          spaceTspan.setAttribute("y", bottom.toString());
+          if((leftNext - right) > 0) spaceTspan.setAttribute("textLength", (leftNext - right).toString());
+          spaceTspan.setAttribute("lengthAdjust", "spacingAndGlyphs");
+          spaceTspan.textContent = " ";
+          paragSvg.appendChild(spaceTspan);
+        }
+
+        // Adds newline at the end of paragraph on Firefox
+        if((i ==  words.length - 1 && (this.onFirefox))) {
+          const nlTspan = document.createElementNS("http://www.w3.org/2000/svg", this.svgWordElement);
+          nlTspan.setAttribute("class", "BRwordElement");
+          nlTspan.setAttribute("x", right.toString());
+          nlTspan.setAttribute("y", bottom.toString());
+          nlTspan.textContent = "\n";
+          paragSvg.appendChild(nlTspan);
```

Comment:
Try this actually; and this should be ok to do always (regardless of browser)
```suggestion
          paragSvg.appendChild(document.createTextNode("\n"));
```

Reply:
done